internal/vercheck/vercheck.go

```diff
@@ -0,0 +1,59 @@
+package vercheck
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/internal/ux"
+	"go.jetpack.io/devbox/internal/xdg"
+	"golang.org/x/mod/semver"
+)
+
+// Keep this in-sync with latest version in launch.sh. If this version is newer
+// Than the version in launch.sh, we'll print a warning.
+const expectedLauncherVersion = "v0.1.0"
+
+func CheckLauncherVersion(w io.Writer) {
+	launcherVersion := os.Getenv("LAUNCHER_VERSION")
+	if launcherVersion == "" {
+		return // Launcher version will not be set in dev.
+	}
+
+	// If launcherVersion is invalid, this will return 0 and we won't print a warning
+	if semver.Compare("v"+launcherVersion, expectedLauncherVersion) < 0 {
+		ux.Fwarning(
+			w,
+			"newer launcher version %s is available (current = v%s), please update "+
+				"using `devbox self-update`\n",
+			expectedLauncherVersion,
+			os.Getenv("LAUNCHER_VERSION"),
+		)
+	}
+}
+
+// SelfUpdate updates the devbox launcher and binary. It ignores and deletes the
+// version cache
+func SelfUpdate(w io.Writer) error {
+	// Delete version cache. Keep this in-sync with whatever logic is in launch.sh
+	cacheDir := xdg.CacheSubpath("devbox")
+	versionCacheFile := filepath.Join(cacheDir, "latest-version")
+	_ = os.Remove(versionCacheFile)
+
+	installScript := "curl -fsSL https://get.jetpack.io/devbox | bash"
+	cmd := exec.Command("sh", "-c", installScript)
+	cmd.Stdout = w
+	cmd.Stderr = w
+	if err := cmd.Run(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	fmt.Fprint(w, "Latest version: ")
+	cmd = exec.Command("devbox", "version")
+	cmd.Stdout = w
+	cmd.Stderr = w
```

Comment:
Oh nice! TIL. This can be used for the bin wrappers as well if we ever move to golang.

Are you sure about setting stdout to stdout? The version print out in this case is purely incidental (because we're using it as a hack to force download the latest binary), I would think that belongs in stderr (which is what `w` is).